OpenSim/Simulation/SimbodyEngine/Test/testJoints.cpp

```diff
@@ -2520,3 +2538,166 @@ void testUniversalJointAccessors()
            __FILE__, __LINE__, "Coordinate accessor failed");
     ASSERT_THROW(OpenSim::InvalidCall, myUniversalJoint.updCoordinate());
 }
+
+void testMotionTypesForCustomJointCoordinates()
+{
+    Model osimModel;
+
+    //OpenSim bodies
+    const Ground& ground = osimModel.getGround();
+    //OpenSim thigh
+    auto osim_thigh = new OpenSim::Body ("thigh", femurMass.getMass(),
+        femurMass.getMassCenter(), femurMass.getInertia());
+
+    // Define hip coordinates and axes for custom joint
+    SpatialTransform hipTransform;
+    OpenSim::Array<std::string> coordNames;
+    coordNames.append("hip_qx");
+    coordNames.append("hip_qy");
+    hipTransform[2].setCoordinateNames(coordNames);
+    hipTransform[2].setFunction(new MultidimensionalFunction());
+    hipTransform[3].setCoordinateNames(OpenSim::Array<std::string>(coordNames[0], 1, 1));
+    hipTransform[3].setFunction(new LinearFunction());
+    hipTransform[4].setCoordinateNames(OpenSim::Array<std::string>(coordNames[1], 1, 1));
+    hipTransform[4].setFunction(new LinearFunction(2.0, -0.5));
+    // define a pure translational dof
+    coordNames.append("hip_tz");
+    hipTransform[5].setCoordinateNames(OpenSim::Array<std::string>(coordNames[2], 1, 1));
+    hipTransform[5].setFunction(new LinearFunction());
+
+    // define a pure rotational dof
+    coordNames.append("hip_rx");
+    hipTransform[0].setCoordinateNames(OpenSim::Array<std::string>(coordNames[3], 1, 1));
+    hipTransform[0].setFunction(new LinearFunction());
+
+    // create custom hip joint
+    auto hip = new CustomJoint("hip", ground, hipInPelvis, SimTK::Vec3(0),
+        *osim_thigh, hipInFemur, SimTK::Vec3(0), hipTransform);
+
+    // Add the thigh body which now also contains the hip joint to the model
+    osimModel.addBody(osim_thigh);
+    osimModel.addJoint(hip);
+
+    // hip_rx is the first coordinate and pure rotational about X
+    auto coordName = hip->getCoordinate(0).getName();
+    auto mt = hip->getCoordinate(0).getMotionType();
+    ASSERT( mt == Coordinate::MotionType::Rotational, __FILE__, __LINE__,
+        "Coordinate `" + coordName + "' failed to register as MotionType::Rotational");
+
+    // hip_qx is the second coordinate coupled to Z rotation but is pure
+    // translational along X
```

Comment:
You are right. (a) more than one coordinate are coupled to affect the motion along a single axis and (b) ...  I could replace "coupled to" with "influencing" since both `hip_qx` and `hip_qy` influence the rotation about the Z axis.